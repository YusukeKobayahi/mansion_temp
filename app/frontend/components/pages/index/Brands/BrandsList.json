{
  "brands": [
    {
      "name": "ザ パーク ハウス",
      "id": "2"
    },
    {
      "name": "シティハウス",
      "id": "3"
    },
    {
      "name": "ライオンズマンション",
      "id": "4"
    },
    {
      "name": "プラウド",
      "id": "5"
    },
    {
      "name": "パークホームズ",
      "id": "6"
    },
    {
      "name": "クレヴィア",
      "id": "7"
    },
    {
      "name": "ブランズ",
      "id": "8"
    },
    {
      "name": "プレミスト",
      "id": "9"
    },
    {
      "name": "パークコート",
      "id": "10"
    },
    {
      "name": "ブリリア",
      "id": "11"
    }
  ]
}
