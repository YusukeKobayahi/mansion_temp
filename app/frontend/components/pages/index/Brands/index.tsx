import Link from "next/link";
import { buildLibraryPagePath } from "~/lib/utils";
import { event } from "~/lib/gtag";
import brandsList from "~/components/pages/index/Brands/BrandsList.json";

import styles from "~/components/pages/index/PopularTowns/index.module.scss";

export type BrandsListProps = {
  name: string;
  id: string;
};

const Prefectures: React.FC = () => {
  const filteredBrandsList: BrandsListProps[] = brandsList.brands;

  const clickBrand = () => {
    event({
      action: "msl_area_click",
      category: "msl_top",
      label: "msl_brands",
      value: 1,
    });
  };

  return (
    <section className={styles.container}>
      <h2 className={styles.title}>マンションブランドから探す</h2>
      <div className={styles.content}>
        <ul className={styles.list}>
          {filteredBrandsList.map((brand) => {
            return (
              <li key={brand.id} className={styles.item}>
                <Link
                  href={buildLibraryPagePath("brands", brand.id)}
                  prefetch={false}
                  passHref
                >
                  <button className={styles.link} onClick={clickBrand}>
                    {brand.name}
                  </button>
                </Link>
              </li>
            );
          })}
        </ul>
      </div>
    </section>
  );
};

export default Prefectures;
