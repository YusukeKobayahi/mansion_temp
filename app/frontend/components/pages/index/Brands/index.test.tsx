import { render } from "@testing-library/react";
import Brands from "~/components/pages/index/Brands";

describe("components/pages/index/Brands", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<Brands />);
    expect(asFragment()).toMatchSnapshot();
  });
});
