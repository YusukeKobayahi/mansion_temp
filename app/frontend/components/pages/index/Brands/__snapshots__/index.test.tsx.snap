// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/index/Brands 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h2
      class="title"
    >
      マンションブランドから探す
    </h2>
    <div
      class="content"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/2"
          >
            ザ パーク ハウス
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/3"
          >
            シティハウス
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/4"
          >
            ライオンズマンション
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/5"
          >
            プラウド
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/6"
          >
            パークホームズ
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/7"
          >
            クレヴィア
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/8"
          >
            ブランズ
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/9"
          >
            プレミスト
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/10"
          >
            パークコート
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/brands/11"
          >
            ブリリア
          </button>
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
